### DISCUSSION TITLE
Search failure in the elasticsearch cluster error help please .. SO version 2.4.40 (logs attached)

### DISCUSSION THREAD

**Discussion by plaotec** [Security-Onion-Solutions/securityonion#12287] [2.4]
PLEASE STOP AND READ THIS INFORMATION!

If you are creating an issue just to ask a question, you will likely get faster and better responses by posting to our discussions forum instead:
https://securityonion.net/discuss

If you think you have found a possible bug or are observing a behavior that you weren't expecting, use the discussion forum to start a conversation about it instead of creating an issue.

If you are very familiar with the latest version of the product and are confident you have found a bug in Security Onion, you can continue with creating an issue here, but please make sure you have done the following:
- duplicated the issue on a fresh installation of the latest version
- provide information about your system and how you installed Security Onion
- include relevant log files
- include reproduction steps

**Comment by plaotec** [comment]
Reference :  https://github.com/Security-Onion-Solutions/securityonion/discussions/11500 

We updated to SO version 2.4.40 but still getting the error:

V2.4.20, we are experiencing a search failure in the elastic cluster. After further review of the logs, it appears that 4 shards failed to initialize kicking a 502 error again

**Comment by plaotec** [comment]
[salt-relay.log](https://github.com/Security-Onion-Solutions/securityonion/files/14105461/salt-relay.log)
[salt-relay-20240116.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105462/salt-relay-20240116.log.gz)
[salt-relay-20240117.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105463/salt-relay-20240117.log.gz)
[salt-relay-20240118.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105464/salt-relay-20240118.log.gz)
[salt-relay-20240119.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105465/salt-relay-20240119.log.gz)
[salt-relay-20240120.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105466/salt-relay-20240120.log.gz)
[salt-relay-20240121.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105467/salt-relay-20240121.log.gz)
[salt-relay-20240122.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105468/salt-relay-20240122.log.gz)
[salt-relay-20240123.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105469/salt-relay-20240123.log.gz)
[salt-relay-20240124.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105470/salt-relay-20240124.log.gz)
[salt-relay-20240125.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105471/salt-relay-20240125.log.gz)
[salt-relay-20240126.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105472/salt-relay-20240126.log.gz)
[salt-relay-20240127.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105473/salt-relay-20240127.log.gz)
[salt-relay-20240128.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105474/salt-relay-20240128.log.gz)
[salt-relay-20240129.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105475/salt-relay-20240129.log.gz)

**Comment by plaotec** [comment]
[Uploading sensoroni-server.zip…]()

**Comment by plaotec** [comment]
[sensoroni-server-20240116.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105484/sensoroni-server-20240116.log.gz)
[sensoroni-server-20240117.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105485/sensoroni-server-20240117.log.gz)
[sensoroni-server-20240118.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105486/sensoroni-server-20240118.log.gz)
[sensoroni-server-20240119.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105487/sensoroni-server-20240119.log.gz)
[sensoroni-server-20240120.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105488/sensoroni-server-20240120.log.gz)
[sensoroni-server-20240121.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105489/sensoroni-server-20240121.log.gz)
[sensoroni-server-20240122.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105491/sensoroni-server-20240122.log.gz)
[sensoroni-server-20240123.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105492/sensoroni-server-20240123.log.gz)
[sensoroni-server-20240124.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105493/sensoroni-server-20240124.log.gz)
[sensoroni-server-20240125.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105495/sensoroni-server-20240125.log.gz)
[sensoroni-server-20240126.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105496/sensoroni-server-20240126.log.gz)
[sensoroni-server-20240127.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105497/sensoroni-server-20240127.log.gz)
[sensoroni-server-20240128.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105498/sensoroni-server-20240128.log.gz)
[sensoroni-server-20240129.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105499/sensoroni-server-20240129.log.gz)

**Comment by plaotec** [comment]
[sync.log](https://github.com/Security-Onion-Solutions/securityonion/files/14105501/sync.log)
[sync-20240116.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105502/sync-20240116.log.gz)
[sync-20240117.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105503/sync-20240117.log.gz)
[sync-20240118.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105504/sync-20240118.log.gz)
[sync-20240119.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105505/sync-20240119.log.gz)
[sync-20240120.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105506/sync-20240120.log.gz)
[sync-20240121.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105507/sync-20240121.log.gz)
[sync-20240122.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105508/sync-20240122.log.gz)
[sync-20240123.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105509/sync-20240123.log.gz)
[sync-20240124.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105510/sync-20240124.log.gz)
[sync-20240125.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105511/sync-20240125.log.gz)
[sync-20240126.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105512/sync-20240126.log.gz)
[sync-20240127.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105513/sync-20240127.log.gz)
[sync-20240128.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105514/sync-20240128.log.gz)
[sync-20240129.log.gz](https://github.com/Security-Onion-Solutions/securityonion/files/14105515/sync-20240129.log.gz)

**Comment by plaotec** [comment]
![error](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/f4a90cb9-b79a-465f-958d-dcf084919d4e)

**Comment by plaotec** [comment]
![deleted](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/eb3538f8-eeef-4329-a1c8-553343bfee6b)

for now, we deleted the indexes with the red health status, and the elasticsearch error went away .. will monitor to see if error comes back after awhile

**Comment by reyesj2** [comment]
run ```sudo so-elasticsearch-query _cat/shards``` any unassigned shards you can look and try to figure out why with the command below (replace .ds-logs-suricata-so-2024.01.30-000001 with whatever shard is showing unassigned)
```
sudo so-elasticsearch-query _cluster/allocation/explain?pretty -d' {  "index": ".ds-logs-suricata-so-2024.01.30-000001", "shard": 0, "primary": true }'
```

What does your grid consist of ? Managersearch + sensor? Manager + sensor(s) + searchnode(s)?

**Comment by plaotec** [comment]
![grid](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/96ce1bd4-cd7d-4749-a0af-1ed5340ee660)


thanks
